NewTimesheet.py

```diff
@@ -0,0 +1,117 @@
+import datetime
+from datetime import date
+import pprint
+import os
+import pandas
+import requests
+
+
+# Set global variables from .env file
+V1_API_URL = "https://api.clockify.me/api/v1"
+WORKSPACE_ID = os.getenv("CLOCKIFY_WORKSPACE_ID")
+HEADERS = {"X-Api-Key": os.getenv("CLOCKIFY_API_KEY")}
+
+def responses_of_url():
+    '''Initial responses of url  '''
+    url = "{}/workspace/{}/users".format(V1_API_URL, WORKSPACE_ID)
+    responses = requests.get(url, headers=HEADERS).json()
+    return responses
+
+def gets_time_sunday():
+    ''' Get the last sunday for the week's range''' 
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday(), weeks=1)
+    sunday = monday + datetime.timedelta(days=6)
+    sunday = sunday.strftime("%Y-%m-%dT23:59:59Z")
+    return sunday
+
+def gets_time_monday():
+    ''' Get the last monday for the  week's range '''
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday(), weeks=1)
+    monday = monday.strftime("%Y-%m-%dT01:00:00Z")
+    return monday
+
+def taking_all_url_members(responses,sunday,monday):
+    ''' Takes the url of each member by id's and the time entries to create dictionarys and input in the data frame '''
+    users = {}
+    dados_frame = pandas.DataFrame(columns = ['project_name', 'client_name', 'task_name', 'user_name', 'description', 'duration'])
+    dic_entry = {}
+    for response in responses:
+        users[response['id']] = response['name']
+    for user_id in users.keys():
+        url_timeentry = "{}/workspaces/{}/user/{}/time-entries?start={}&end={}&hydrated={}".format(V1_API_URL, WORKSPACE_ID, user_id , monday, sunday, True)
+        timeentries_member = requests.get(url_timeentry, headers=HEADERS).json()
+        for time_entry in timeentries_member:
+            try:
+                client_name = time_entry['tags'][0]['name']
+            except IndexError and ValueError:
+                print(time_entry)
+                continue
+            end = time_entry['timeInterval']['end']
+            end = datetime.datetime.strptime(end, '%Y-%m-%dT%H:%M:%SZ')
+            start = time_entry['timeInterval']['start']
+            start = datetime.datetime.strptime(start, '%Y-%m-%dT%H:%M:%SZ')            
+            duration = ((end-start).total_seconds())/3600
+            dic_entry['task_name'] = time_entry['task']['name']
+            dic_entry['project_name'] = time_entry['project']['name']
+            dic_entry['client_name'] = client_name
+            dic_entry['user_name'] = users[time_entry['userId']]
+            dic_entry['description'] = time_entry['description']
+            dic_entry['duration'] = duration
+            dados_frame = dados_frame.append(dic_entry, ignore_index=True)
+    print (dados_frame)
+    return dados_frame
+
+
+def Organize_Data_Frame_create_excel (dados_frame):
+    '''' Organize D.Frame and create a Excel'''
```

Comment:
I would be more specific on what are you actually organizing and separate the part that creates the Excel in another function.

```suggestion
    """Create 'Horas-aulas' column and group data based on project, task, user, and client"""
```